packages/angular_devkit/build_angular/src/webpack/configs/typescript.ts

```diff
@@ -13,9 +13,73 @@ import {
   AngularCompilerPlugin,
   AngularCompilerPluginOptions,
   NgToolsLoader,
-  PLATFORM
+  PLATFORM,
+  ivy,
 } from '@ngtools/webpack';
 import { WebpackConfigOptions, BuildOptions } from '../../utils/build-options';
+import { CompilerOptions } from '@angular/compiler-cli';
+import { RuleSetLoader } from 'webpack';
+
+function canUseIvyPlugin(wco: WebpackConfigOptions): boolean {
+  // Can only be used with Ivy
+  if (!wco.tsConfig.options.enableIvy) {
+    return false;
+  }
+
+  // Allow fallback to legacy build system via environment variable ('NG_BUILD_IVY_LEGACY=1')
+  const flag = process.env['NG_BUILD_IVY_LEGACY'];
```

Comment:
Can we place this in `packages/angular_devkit/build_angular/src/utils/environment-options.ts`? It's easies to keep track if these.